### ISSUE TITLE
Multiple Config rules using same Lambda returns 'There is already a Construct' name error

### ISSUE THREAD

**Issue by mcmluk123** [aws/aws-cdk#9756]
I am trying to deploy multiple custom configuration rules which use the same lambda, but different input parameters. However, I get a duplicate construct error:

> There is already a Construct with name 'Permission' in Function [amiComplianceFn]

### Reproduction Steps
``` ts
    //Create AMI compliance Lambda
    const amiComplianceFn = new lambda.Function(this, 'amiComplianceFn',{
      runtime: lambda.Runtime.NODEJS_10_X,  // execution environment
      code: lambda.Code.fromAsset('lambda'),  // code loaded from "lambda" directory
      handler: 'ami_lambda.handler',  // file is "report", function is "handler"
      timeout: cdk.Duration.seconds(15)
    });
    //Add Managed policy to Lambda function to allow query aws config amd EC2
    amiComplianceFn.role?.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2FullAccess'));
    const amiCompliance = new awscfg.CustomRule(this, 'amiCompliance', {
      lambdaFunction: amiComplianceFn,
      configRuleName: `ampc-ami-30-days`,
      inputParameters: { amiExpiryDay: 30 },
      configurationChanges: true
    });
    amiCompliance.scopeToResources('AWS::EC2::Instance');

    const amiCompliance15 = new awscfg.CustomRule(this, 'amiCompliance15', {
      lambdaFunction: amiComplianceFn,
      configRuleName: `ampc-ami-15-days`,
      inputParameters: { amiExpiryDay: 15 },
      configurationChanges: true
    });
    amiCompliance15.scopeToResources('AWS::EC2::Instance');
    
    const amiCompliance25 = new awscfg.CustomRule(this, 'amiCompliance25', {
      lambdaFunction: amiComplianceFn,
      configRuleName: `ampc-ami-25-days`,
      inputParameters: { amiExpiryDay: 25 },
      configurationChanges: true
    });
    amiCompliance25.scopeToResources('AWS::EC2::Instance');
```

### What did you expect to happen?
Multiple config rules created using the same lambda

### What actually happened?
Error regarding duplicate construct :

> There is already a Construct with name 'Permission' in Function [amiComplianceFn]

### Environment

  - **CLI Version      :** 1.53.0 (build 6c326cb)
  - **Framework Version:**
  - **Node.js Version:** v12.18.2
  - **OS               :**
  - **Language (Version):** 

### Other

--- 

This is :bug: Bug Report

**Comment by SomayaB** [comment]
Related to #8553

**Comment by nija-at** [comment]
This should be fixed in https://github.com/aws/aws-cdk/commit/0fc5899364b8fb3b2fac8cb774f103054607bb2e which has been released in `1.61.0`. 

Please upgrade to this version and see if this fixes the issue for you.
